src/ctap2/commands/get_assertion.rs

```diff
@@ -456,17 +452,11 @@ impl GetAssertionResult {
         rp_id_hash: &RpIdHash,
         key_handle: &PublicKeyCredentialDescriptor,
     ) -> Result<GetAssertionResult, CommandError> {
-        let parse_authentication = |input| {
-            // Parsing an u8, then a u32, and the rest is the signature
-            let (rest, (user_presence, counter)) = tuple((be_u8, be_u32))(input)?;
-            let signature = Vec::from(rest);
-            Ok((user_presence, counter, signature))
-        };
-        let (user_presence, counter, signature) =
-            parse_authentication(input).map_err(|e: nom::Err<VerboseError<_>>| {
-                error!("error while parsing authentication: {:?}", e);
-                CommandError::Deserializing(DesError::custom("unable to parse authentication"))
-            })?;
+        let mut data = Cursor::new(input);
+        let user_presence = read_byte(&mut data).map_err(CommandError::Deserializing)?;
+        let counter = read_be_u32(&mut data).map_err(CommandError::Deserializing)?;
+        // Remaining data is signature (Note: `data.remaining_slice()` is not yet stabilized)
+        let signature = Vec::from(&data.get_ref()[data.position() as usize..]);
```

Comment:
Do we need a bounds check before indexing with `data.position()`?

Reply:
No. We cannot move the cursor past it's end by reading (we could, when using `set_position()`, which we don't use). Either we error out before, because not enough data was there for the u32, or we will have a range from "end to end", meaning an empty range, so we will get an empty slice. See [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=15f3f584249157b337561f5905ec2c85) as an example.

Reply:
ok!